rewrite-java/src/main/java/org/openrewrite/java/cleanup/FinalizePrivateFields.java

```diff
@@ -49,8 +49,12 @@ protected TreeVisitor<?, ExecutionContext> getVisitor() {
             private Set<JavaType.Variable> privateFieldsToBeFinalized = new HashSet<>();
 
             @Override
-            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl,
-                ExecutionContext ctx) {
+            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                if (!classDecl.getLeadingAnnotations().isEmpty()) {
+                    // skip if a class has any annotation, since the annotation could generate some code to assign
+                    // fields like Lombok @Setter.
+                    return classDecl;
```

Comment:
Should we [add `super.visitClassDeclaration` at the top](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices#be-deliberate-about-lst-traversal) here to check for inner classes?
Seems like we could potentially miss quite a few cases if we return early for any class with any annotation.

Reply:
I intended to skip the entire class including any inner class or static nested class if an outer class has any annotations.
since I think the outer class is the majority we want to cover in this recipe.
my concern was skipping outer class but with inner class covered could be risky, not sure if annotations can do something for the inner classes. so prefer to skip the entire class including the inner class to make sure it's safe.

Reply:
Perfect, then let's start as is. :)